morph_tool/morphdb.py

```diff
@@ -0,0 +1,371 @@
+        Raises: ValueError if the neurondb does not abide by the specification
+        https://bbpteam.epfl.ch/documentation/projects/morphology-repair-workflow/latest/input_files.html#specification
+
+        ..note:: missing keys are filled with `True` values
+        '''
+        obj = MorphDB()
+
+        if not morphology_folder:
+            morphology_folder = neurondb.parent.resolve()
+
+        morph_paths = {path.stem: path for path in iter_morphology_files(morphology_folder)}
+
+        if neurondb.suffix.lower() == '.dat':
+            columns = ['name', 'layer', 'mtype']
+            obj.df = pd.read_csv(neurondb, sep=r'\s+', names=columns, usecols=range(len(columns)))
+
+            fulltypes = obj.df.mtype.str.split(':', n=1, expand=True).fillna('')
+            if len(fulltypes.columns) > 1:
+                obj.df['mtype_no_subtype'] = fulltypes[0]
+                obj.df['msubtype'] = fulltypes[1]
+            else:
+                obj.df['mtype_no_subtype'] = obj.df.mtype
+                obj.df['msubtype'] = ''
+            obj.df['label'] = label
+            for missing_col in set(COLUMNS) - set(obj.df.columns):
+                obj.df[missing_col] = None
+            obj.df.layer = obj.df.layer.astype('str')
+            obj.df['path'] = obj.df.name.map(morph_paths)
+            obj.df = obj.df.reindex(columns=COLUMNS)
+            for key in BOOLEAN_REPAIR_ATTRS:
+                obj.df[key] = True
+            obj.df['axon_inputs'] = [[] for _ in range(len(obj.df))]
+
+            return obj
+
+        with neurondb.open() as fd:
+            content = fd.read()
+            neurondb = xmltodict.parse(content)
+
+        morphologies = neurondb['neurondb']['listing']['morphology']
+
+        # Case where there is a single <morphology></morphology> tag in the XML
+        if not isinstance(morphologies, list):
+            morphologies = [morphologies]
+        morphologies = filter(None, morphologies)
+        morphologies = list(map(MorphInfo._from_xmldict, morphologies))  # noqa, pylint: disable=protected-access
+
+        for morph in morphologies:
+            morph.label = label
+            morph.path = morph_paths.get(morph.name)
+
+        obj.df = MorphDB._create_dataframe(morphologies)
+        obj.df = obj.df.astype({key: bool for key in BOOLEAN_REPAIR_ATTRS})
+        return obj
+
+    @classmethod
+    def from_folder(cls,
+                    morphology_folder: Path,
+                    mtypes: Sequence[Tuple[str, str]],
+                    label: str = 'default'):
+        '''Factory method to create a MorphDB object from a folder containing morphologies
+
+        Args:
+            morphology_folder: a folder containing morphologies
+            mtypes: a sequence of 2-tuples (morphology name, mtype)
+            label: (optional) a group label to be used to identify the morphlogies from this folder
+        '''
+
+        paths = {path.stem: path for path in iter_morphology_files(morphology_folder)}
```

Comment:
If I'm not wrong, a morphology folder can contain morphologies in different formats ('asc', 'h5', 'swc'), but we are keeping only one path (the last one returned by iter_morphology_files).
Is it ok? In that case, could you add a comment or an explanation in the docstring?